### PR TITLE
test: fix plugin test to work with forks and feature branches

### PR DESCRIPTION
**Pull request by yousfiSaad**
## Summary

Fixes the plugin test compatibility issue that prevented MegaLinter tests from working correctly on forked repositories and feature branches.

## Problem

The test `test_load_plugin_success()` was hardcoded to fetch plugin descriptors from `oxsecurity/megalinter/main`, causing test failures for contributors working on forks or feature branches.

```python
# Before: hardcoded values
local_branch = "main"
repo_slug = ML_REPO  # "oxsecurity/megalinter"
```

This meant:
- Forks (`username/megalinter`) still tried to load from `oxsecurity/megalinter`
- Feature branches (`my-feature`) still tried to load from `main`
- Tests failed with HTTP 404 errors

## Solution

Added dynamic git repository and branch detection using `gitpython`:

```python
# After: dynamic detection
repo_slug, local_branch = get_git_repo_info()
```

The `get_git_repo_info()` function:
- Detects the current git repository from `__file__`
- Gets the current branch name via `repo.active_branch.name`
- Parses the remote URL to extract `owner/repo`
- Falls back to `(ML_REPO, "main")` if detection fails

## Changes

- **File**: `megalinter/tests/test_megalinter/plugins_test.py`
- **Added**: `get_git_repo_info()` helper function (43 lines)
- **Updated**: `test_load_plugin_success()` to use dynamic repo/branch detection
- **Removed**: Hardcoded `local_branch = "main"` and old TODO comment

## Impact

| Before | After |
|--------|-------|
| ❌ Tests fail on forks | ✅ Tests use fork's remote |
| ❌ Tests fail on feature branches | ✅ Tests use current branch |
| ❌ Contributors must modify tests | ✅ Tests just work |
| ❌ High barrier to contribution | ✅ Low barrier to contribution |

## Testing

```bash
# Run the specific test
python -m pytest megalinter/tests/test_megalinter/plugins_test.py::plugins_test::test_load_plugin_success -v
```

## Documentation

Comprehensive documentation added in `FIX_DOCUMENTATION.md` (847 lines):
- Beginner-friendly introduction to MegaLinter concepts
- Complete real-world example with "Alex's contributor journey"
- Step-by-step code walkthrough
- Edge case analysis

## Checklist

- [x] Code follows project conventions
- [x] Documentation added/updated
- [x] Tests pass locally
- [x] No breaking changes

---

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> Improves plugin test robustness by detecting the current repository slug and branch at runtime, enabling successful runs on forks and non-main branches.
> 
> - Adds `get_git_repo_info()` with helpers to parse Git remotes, CI env (`GITHUB_HEAD_REF`, `GITHUB_REF_NAME`), and fork PR refs
> - Updates `test_load_plugin_success()` to use detected `repo_slug` and `branch`, URL-encoding branch via `quote()`
> - Removes hardcoded `ML_REPO/main` usage; falls back to `(ML_REPO, "main")` on failure
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a8b9089ac43411c34414757f88e797b869feb034. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: megalinter/tests/test_megalinter/plugins_test.py [hunk 1]
```diff
@@ -8,10 +8,129 @@
...
+    """
+    Get branch name from CI environment variables.
+    Uses the same priority order as megalinter/utils.py get_git_context_info().
+    Returns None if no CI environment variable is set.
+    """
+    return (
+        os.environ.get("GITHUB_HEAD_REF")
+        or os.environ.get("GITHUB_REF_NAME")
+        or os.environ.get("GIT_BRANCH")
+        or os.environ.get("CI_COMMIT_REF_NAME")
```
Comment (nvuillam) [inline megalinter/tests/test_megalinter/plugins_test.py]: @yousfiSaad why using variables of other git providers (gitlab, bitbucket...) ?

Comment (yousfiSaad) [reply megalinter/tests/test_megalinter/plugins_test.py]: @nvuillam Good point! I copied the pattern from `utils.py:get_git_context_info()` for consistency, but you're right — since this test specifically constructs GitHub URLs (`raw.githubusercontent.com`), non-GitHub CI variables don't make sense here. I can simplify to GitHub-only variables. I'll fix it
